mitosheet/mitosheet/public/v3/formatting.py

```diff
@@ -8,19 +8,43 @@
 
 from mitosheet.excel_utils import get_column_from_column_index
 
+# Object to map the conditional formatting operators to the openpyxl operators
+CONDITIONAL_TO_OPENPYXL_OPERATOR_MAP = {
+    'greater': 'greaterThan',
+    'less': 'lessThan',
+    'number_exactly': 'equal',
+    'number_not_exactly': 'notEqual',
+    'greater_than_or_equal': 'greaterThanOrEqual',
+    'less_than_or_equal': 'lessThanOrEqual',
+}
+
 def add_conditional_formats(
     conditional_formats: list,
     sheet: Worksheet,
     df: DataFrame
 ) -> None:
     for conditional_format in conditional_formats:
-        for filter in conditional_format.get('filters'):
+        for filter in conditional_format.get('filters', []):
             # Start with the greater than condition
-            if filter['condition'] != 'greater':
+            operator = CONDITIONAL_TO_OPENPYXL_OPERATOR_MAP.get(filter['condition'])
+            if operator is None:
+                continue
+            cond_fill = None
+            cond_font = None
+            if conditional_format.get('background_color') is not None:
+                cond_fill = PatternFill(start_color=conditional_format['background_color'][1:], end_color=conditional_format['background_color'][1:], fill_type='solid')
+            if conditional_format.get('font_color') is not None:
+                cond_font = Font(color=conditional_format['font_color'][1:])
+            
+            if cond_fill is None and cond_font is None:
                 continue
-            cond_fill = PatternFill(start_color=conditional_format['background_color'][1:], end_color=conditional_format['background_color'][1:], fill_type='solid')
-            cond_font = Font(color=conditional_format['font_color'][1:])
-            column_conditional_rule = CellIsRule(operator="greaterThan", font=cond_font, fill=cond_fill, formula=[f'{filter["value"]}'])
+            elif cond_fill is None:
+                column_conditional_rule = CellIsRule(operator=operator, font=cond_font, formula=[f'{filter["value"]}'])
+            elif cond_font is None:
+                column_conditional_rule = CellIsRule(operator=operator, fill=cond_fill, formula=[f'{filter["value"]}'])
+            else:
+                column_conditional_rule = CellIsRule(operator=operator, fill=cond_fill, font=cond_font, formula=[f'{filter["value"]}'])
+            
```

Comment:
I think these cases can be collapsed into just two cases. The way that optional arguments work in Python is that they are most often set to be equal to None by default. In this case, looking at the CellIsRule, we have the definition:
```
def CellIsRule(operator=None, formula=None, stopIfTrue=None, font=None, border=None, fill=None):
```

Thus, if the cond_fill is None, and you pass it to the file -- this will be like you didn't pass it at all (as the default is None). 

So I think all of this can be collapsed into two cases - one where both cond_fill and cond_font are None, and the other case where you just pass both arguments.

Reply:
Ah true! Just added